### PR TITLE
test: fix the crp watcher integration test flakiness

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

make sure the event has arrived before resetting the queue to fix the test flakiness

Fixes #

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

run the unit test ci

### CODE REVIEW

File: pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go [hunk 1]
```diff
@@ -82,6 +82,13 @@ var _ = Describe("Test ClusterResourcePlacement Watcher", func() {
 
 	Context("When updating clusterResourcePlacement", func() {
 		BeforeEach(func() {
+			By("By checking the placement queue before resetting")
+			// The event could arrive after the resetting, which causes the flakiness.
+			// It makes sure the queue is clear before proceed.
+			Eventually(func() bool {
+				return fakePlacementController.Key() == testCRPName
+			}, eventuallyTimeout, interval).Should(BeTrue(), "placementController should receive the CRP name")
```
Comment (ryanzhang-oss) [inline pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go:90]: I don't fully get it. If we want to get a clean start for all the test, why create the CPR in the first place at line 60 and then reset it before each test?

Comment (ryanzhang-oss) [reply pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go:90]: one can move the create to the test in line 75

Comment (ryanzhang-oss) [reply pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go:90]: and here

Comment (ryanzhang-oss) [reply pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go:90]: or move the test in line 75 to the BeforeEach for all.

Comment (zhiying-lin) [reply pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go]: moved the create test into the BeforeEach.

**Review by michaelawyu — APPROVED** [review]
LGTM.

File: pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go [hunk 2]
```diff
@@ -58,33 +58,19 @@ var _ = Describe("Test ClusterResourcePlacement Watcher", func() {
...
 	})
 
 	Context("When updating clusterResourcePlacement", func() {
-		BeforeEach(func() {
-			By("By resetting the placement queue")
-			fakePlacementController.ResetQueue()
-		})
 		AfterEach(func() {
 			By("By deleting crp")
 			createdCRP = clusterResourcePlacementForTest()
```
Comment (ryanzhang-oss) [inline pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go:76]: why creating this again?

File: pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go [hunk 3]
```diff
@@ -58,33 +58,19 @@ var _ = Describe("Test ClusterResourcePlacement Watcher", func() {
 		By("By creating a new clusterResourcePlacement")
 		crp := clusterResourcePlacementForTest()
```
Comment (ryanzhang-oss) [inline pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go:59]: why not assign this to createdCRP so we don't need to create it again when we delete it?

**Review by ryanzhang-oss — APPROVED** [review]

File: pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go (hunk 2 above)
Comment (zhiying-lin) [reply pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go:76]: oh, it's not creating a new CRP by sending to the API server and it's building a CRP structure. the name is not good. Will rename this variable.

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**